app/Models/Post.php

```diff
@@ -277,12 +277,16 @@ public function tag($tag)
 
         $tag = Tag::firstOrCreate(['tag_name' => $tagName], ['tag_slug' => str_slug($tagName, '-')]);
 
-        $this->tags()->attach($tag);
+        // Only tag if the tag doesn't exist on the post yet.
+        // Otherwise, an integrity constraint violation / duplicate entry error will be thrown.
+        if (! $this->tagNames()->contains($tagName)) {
```

Comment:
for some reason, adding this check here breaks the tests but I'm not quiet sure why! Still trying to dig into this...

Reply:
does `tagNames()` return a collection or an array? I'm wondering if the `contains()` method works here?

Reply:
a collection! @sbsmith86

Reply:
Also do you see a more verbose error in the logs when the test fails?

Reply:
hmmm I think I need to pull down this branch to check it out.

Reply:
ty! I'll keep digging too - this is a 👻 -y one!

Reply:
also what happens if you ``dd$this->tagNames()->contains($tagName)`` before the if- statement.

Reply:
@sbsmith86 doing that returns `false` (and `dd(! $this->tagNames()->contains($tagName));` returns `true`!).

Reply:
wait - this is ringing a bell that I think I had trouble with before and could only get tests to pass by tagging via the API? I'm gonna do some digging through :slack: and :github: to see what the solution was!

Reply:
I found [the comment](https://github.com/DoSomething/rogue/pull/697#issuecomment-389298124) I was thinking of but I don't think this actually has anything to do with what's going on here :-/

Reply:
Even tagging with the API is breaking: 
```
$this->withAdminAccessToken()->postJson('api/v3/posts/' . $post->id . '/tags', [
             'tag_name' => 'Good Submission',
        ]);

        $this->withAdminAccessToken()->postJson('api/v3/posts/' . $post->id . '/tags', [
             'tag_name' => 'Tag To Delete',
        ]);
``` 
and it seems as it is just this new line that the test doesn't like 🤔 
`        // if (! $this->tagNames()->contains($tagName)) {`